eva-server/src/test/java/uk/ac/ebi/eva/server/ws/VariantWSServerV2IntegrationTest.java

```diff
@@ -0,0 +1,175 @@
+    public void rootTestGetVariantsByVariantId() throws URISyntaxException {
+        String url = "/v2/variants/20:60100:A:T?species=mmusculus&assembly=grcm38";
+        List<VariantWithSamplesAndAnnotation> variantWithSamplesAndAnnotations = WSTestHelpers.testRestTemplateHelper(
+                url,restTemplate);
+        assertTrue(variantWithSamplesAndAnnotations.size() > 0);
+        assertTrue(variantWithSamplesAndAnnotations.get(0).getSourceEntries().size() == 0);
+        assertNull(variantWithSamplesAndAnnotations.get(0).getAnnotation());
+        assertTrue(variantWithSamplesAndAnnotations.get(0).getIds().size() > 0);
+    }
+
+    @Test
+    public void rootTestGetVariantsByNonExistingVariantId() throws URISyntaxException {
+        String url = "/v2/variants/10:0:A:T?species=mmusculus&assembly=grcm38";
+
+        ResponseEntity<QueryResponse<QueryResult<VariantWithSamplesAndAnnotation>>> response = restTemplate.exchange(
+                url, HttpMethod.GET, null,
+                new ParameterizedTypeReference<QueryResponse<QueryResult<VariantWithSamplesAndAnnotation>>>() {
+                });
+        assertEquals(HttpStatus.NOT_FOUND, response.getStatusCode());
+        assertEquals(0, response.getBody().getResponse().size());
+    }
+
+    @Test
+    public void rootTestForError() throws URISyntaxException {
+        String url = "/v2/variants/13:32889669:C:T?species=&assembly=grcm38";
+        assertEquals("Please specify a species", testForErrorHelper(url));
+    }
+
+    private String testForErrorHelper(String url) {
+        return WSTestHelpers.testRestTemplateHelperForError(url, restTemplate);
+    }
+
+    @Test
+    public void annotationEndPointTestExisting() throws URISyntaxException {
+        String url = "/v2/variants/20:60100:A:T/annotations?species=mmusculus&assembly=grcm38";
+        ResponseEntity<QueryResponse<QueryResult<Annotation>>> annotations = restTemplate.exchange(
+                url, HttpMethod.GET, null,
+                new ParameterizedTypeReference<QueryResponse<QueryResult<Annotation>>>() {
+                });
+        assertEquals(HttpStatus.OK, annotations.getStatusCode());
+        assertFalse(annotations.getBody().getResponse().get(0).getResult().get(0).getChromosome().isEmpty());
+    }
+
+    @Test
+    public void annotationEndPointTestNonExisting() throws URISyntaxException {
+        String url = "/v2/variants/100:0:C:T/annotations?species=mmusculus&assembly=grcm38";
+        ResponseEntity<QueryResponse<QueryResult<Annotation>>> annotations = restTemplate.exchange(
+                url, HttpMethod.GET, null,
+                new ParameterizedTypeReference<QueryResponse<QueryResult<Annotation>>>() {
+                });
+        assertEquals(HttpStatus.NOT_FOUND, annotations.getStatusCode());
+        assertTrue(annotations.getBody().getResponse().get(0).getResult().size() == 0);
+    }
+
+    @Test
+    public void annotationEndpointTestForError() throws URISyntaxException {
```

Comment:
Add another assert for the case when the cache version is provided but the VEP version is not.

Reply:
Added :)